compose/cli/log_printer.py

```diff
@@ -230,7 +235,7 @@ def consume_queue(queue, cascade_stop):
 
         if item.is_stop:
             if cascade_stop:
-                raise StopIteration
+                yield item.item
```

Comment:
we already `yield item.item` unconditionally below, so it seems like this whole block can be changed to
```
if item.is_stop and not cascade_stop:
  continue
```